routes/auth.routes.js

```diff
@@ -0,0 +1,125 @@
+const { Router } = require("express");
+const router = new Router();
+
+const mongoose = require("mongoose"); 
+const bcryptjs = require("bcryptjs");
+const saltRounds = 10;
+
+const User = require("../models/User.model");
+
+// GET route ==> to display the signup form to users
+router.get("/signup", (req, res) => res.render("auth/signup"));
+
+// POST route ==> to process form data
+router.post("/signup", (req, res, next) => {
+  // console.log("The form data: ", req.body);
+
+  const { username, email, password } = req.body;
+
+  // make sure users fill all mandatory fields:
+  if (!username || !email || !password) {
+    res.render("auth/signup", {
+      errorMessage: "All fields are mandatory. Please provide your username, email and password."
+    });
+    return;
+  }
+
+  // make sure passwords are strong:
+  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+  if (!regex.test(password)) {
+    res.status(500).render("auth/signup", {
+      errorMessage:
+        "Password needs to have at least 6 chars and must contain at least one number, one lowercase and one uppercase letter."
+    });
+    return;
+  }
+
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({
+        // username: username
+        username,
+        email,
+        // passwordHash => this is the key from the User model
+        //     ^
+        //     |            |--> this is placeholder (how we named returning value from the previous method (.hash()))
+        passwordHash: hashedPassword
+      });
+    })
+    .then((userFromDB) => {
+      console.log("Newly created user is: ", userFromDB);
+      res.redirect("/userProfile");
```

Comment:
not really a mistake, but if you do not create a session after creating the user, redirecting to the profile only to show them an error is not a really good user experience. It would be best to redirect them to the login page instead